src/createAtomCreators.ts

```diff
@@ -1,22 +1,30 @@
-import { createTRPCClient } from '@trpc/client';
+import { createTRPCClient, TRPCRequestOptions } from '@trpc/client';
 import type { CreateTRPCClientOptions } from '@trpc/client';
-import type { AnyRouter } from '@trpc/server';
+import type { AnyRouter, inferHandlerInput } from '@trpc/server';
 
 import { atom } from 'jotai';
 import type { Getter } from 'jotai';
 
 const isGetter = <T>(v: T | ((get: Getter) => T)): v is (get: Getter) => T =>
   typeof v === 'function';
 
+type ArgsOrGetter<T> = T | ((get: Getter) => T);
+
 export function createAtomCreators<TRouter extends AnyRouter>(
   opts: CreateTRPCClientOptions<TRouter>,
 ) {
   const client = createTRPCClient<TRouter>(opts);
 
-  type Args = Parameters<typeof client.query>;
-
-  const atomWithQuery = (
-    getArgs: Args | ((get: Getter) => Args),
+  type TQueries = TRouter['_def']['queries'];
+  const atomWithQuery = <
+    TPath extends keyof TRouter['_def']['queries'] & string,
```

Comment:
Can it be this?
```suggestion
    TPath extends keyof TQueries & string,
```

Reply:
In v10 we moved to a proxy API so it would look more like this:
```ts
atomCreator.greeting.atomWithQuery("hello")
atomCreator.post.byId.atomWithQuery({ id: 2 })
```

Reply:
Oooops, I wasn't aware that by looking at the docs.
That's going to be very tricky. Not sure how DX goes...

Reply:
DX is alright. I implemented it for our server caller, client, and React API and it wasn't too bad.

Reply:
I mean, DX with atoms. I'd stick with the function style.
Can we keep using `createTRPCClient`, or is it going to be deprecated?

Reply:
It's deprecated for v10 so users switch to the proxy API although all our integrations still use it under the hood.

Reply:
Can we use it like this?
```ts
const client = createTRPCProxyClient<AppRouter>({ ... });

  const path = 'getUser'; // from function argument
  await client[path].query('id_bilbo');
```

Also, how would `.post.byId` be translated for `createTRPCClient`?

Reply:
I don't think it's working right now but ideally integrations could just do `client.query(path, 'id_bilbo')`.

Reply:
> integrations could just do `client.query(path, 'id_bilbo')`.

Do you mean `createTRPCProxyClient` will also allow `client.query(path, 'id_bilbo')`?

To be honest, integrations (including 3rd-party ones) would never need Proxies from vanilla client, so `createTRPCClient` would make more sense (as it's probably smaller.)

Reply:
That's a good point. I'll bring it up and we'll probably expose it via a different API.